packages/x-components/src/plugins/__tests__/x-plugin-mixin.spec.ts

```diff
@@ -26,54 +28,91 @@ describe('testing $x component API global mixin', () => {
-
-      @Component({
-        template: `<div><slot/></div>`,
-        provide: {
-          origin: 'origin-test'
-        }
-      })
-      class Provider extends Vue {}
+    interface RenderOriginComponentOptions {
+      feature?: Feature;
+      location?: FeatureLocation;
+    }
 
-      @Component({ template: '<div>{{ $origin }}</div>' })
-      class Injector extends Vue {}
+    interface RenderOriginComponentAPI {
+      busListener: jest.SpyInstance;
+      wrapper: Wrapper<Vue>;
+    }
 
+    function renderOriginComponent({
+      feature,
+      location
+    }: RenderOriginComponentOptions = {}): RenderOriginComponentAPI {
+      const busListener = jest.spyOn(XPlugin.bus, 'emit');
       const wrapper = mount(
         {
-          template: `<Provider><Injector /></Provider>`,
+          template: `
+            <MyButton/>`,
           components: {
-            Provider,
-            Injector
+            MyButton: {
+              template: '<button @click="emit">Click me</button>',
+              methods: {
+                emit(this: Vue) {
+                  this.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill', {
+                    feature
+                  });
+                }
+              }
+            }
+          },
+          provide: {
+            location
           }
-        } as ComponentOptions<any> & ThisType<Vue>,
+        },
         {
           localVue
         }
       );
+      return { wrapper, busListener };
+    }
+
+    it("doesn't include origin in the $x.emit if there is no location", () => {
+      const { wrapper, busListener } = renderOriginComponent({
+        feature: 'search_box',
+        location: undefined
+      });
+      wrapper.trigger('click');
+
+      expect(busListener).toHaveBeenCalledTimes(1);
+      expect(busListener).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.anything(),
+        expect.objectContaining<Partial<WireMetadata>>({ origin: undefined })
+      );
+    });
+
+    it("doesn't include origin in the $x.emit if there is no feature", () => {
+      const { wrapper, busListener } = renderOriginComponent({
+        feature: undefined,
+        location: 'predictive_layer'
+      });
+      wrapper.trigger('click');
+
+      expect(busListener).toHaveBeenCalledTimes(1);
+      expect(busListener).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.anything(),
+        expect.objectContaining<Partial<WireMetadata>>({ origin: undefined })
+      );
+    });
 
-      wrapper
-        .findComponent(Injector)
-        .vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
-      componentInstance.vm.$x.on('UserIsTypingAQuery', true).subscribe(listener);
+    it('emits a valid origin when both location and feature are available', () => {
+      const { wrapper, busListener } = renderOriginComponent({
+        feature: 'search_box',
+        location: 'predictive_layer'
```

Comment:
I'm not sure what is the syntax for this `location` as I can see both `location: 'empty-search'`(in the `bus.decorators.spec.ts`  file) and ` location: 'predictive_layer'`.

Reply:
`empty-search` shouldn't exist. Fixed it.